src/codegen/type/type_system.cpp

```diff
@@ -210,11 +251,34 @@ Value TypeSystem::ExpensiveComparisonHandleNull::EvalCompareGte(
 
 Value TypeSystem::ExpensiveComparisonHandleNull::EvalCompareForSort(
     CodeGen &codegen, const Value &left, const Value &right) const {
-  auto impl = [this](CodeGen &codegen, const Value &left, const Value &right) {
+  // If neither input is NULLable, we're done
+  if (!left.IsNullable() && !right.IsNullable()) {
     return CompareForSortImpl(codegen, left, right);
-  };
-  const auto &result_type = Integer::Instance();
-  return GenerateBinaryHandleNull(codegen, result_type, left, right, impl);
+  }
+
+  /// The logic here is very similar to
+  /// SimpleComparisonHandleNull::EvalCompareForSort() except we **generate**
+  /// an if-clause becuase the non-null check is unsafe.
```

Comment:
Just for clarification: The difference to *SimpleComparisonHandleNull* is, that calling `CompareForSortImpl` in this case is unsafe if one of the values is null?

Reply:
Correct.